Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -153,32 +151,79 @@ public function deleteCookie(string $cookieName): void
      * @param string $parameterName
      * @param mixed $value
      * @return void
-     * @api
+     * @deprecated since Flow 7.0 use setHttpHeader or replaceHttpResponse instead. See also #linkToDocs
      */
     public function setComponentParameter(string $componentClassName, string $parameterName, $value): void
     {
-        if (!isset($this->componentParameters[$componentClassName])) {
-            $this->componentParameters[$componentClassName] = [];
+        if ($componentClassName === SetHeaderComponent::class) {
+            $this->setHttpHeader($parameterName, $value);
+            return;
         }
-        $this->componentParameters[$componentClassName][$parameterName] = $value;
+
+        if ($componentClassName === ReplaceHttpResponseComponent::class && $parameterName === 'response') {
+            $this->replaceHttpResponse($value);
+            return;
+        }
+        throw new \InvalidArgumentException('todo');
```

Comment:
```suggestion
        throw new \InvalidArgumentException('setComponentParameters is deprecated and will not work with anything other than SetHeaderComponent and ReplaceHttpResponseComponent argument. If you want to send data to your middleware from the action, use response headers or introduce a global context. Both solutions are to be considered bad practice though.');
```
Wording @bwaidelich @sorenmalling ?

Reply:
What about
```suggestion
        throw new \InvalidArgumentException(sprintf('The method %s is deprecated. It will only allow a $componentClassName parameter of "%s" or "%s". If you want to send data to your middleware from the action, use response headers or introduce a global context. Both solutions are to be considered bad practice though.', __METHOD__, SetHeaderComponent::class, ReplaceHttpResponseComponent::class), 1605088079);
```